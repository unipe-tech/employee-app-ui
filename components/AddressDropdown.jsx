import { useEffect, useState, memo } from "react";
import { useDispatch, useSelector } from "react-redux";

import { addESICAddress } from "../store/slices/esicSlice";
import FormInput from "./atoms/FormInput";
import DropDownForm from "./molecules/DropDownForm";

const customData = require("../assets/state_districts.json");

const AddressDropdown = ({ type }) => {
  const [districts, setDistricts] = useState(["Please Choose a State"]);

  const dispatch = useDispatch();
  const states = Object.keys(customData);

  const [geoState, setGeoState] = useState(
    useSelector((state) => state.esic.address[type].state)
  );
  const [district, setDistrict] = useState(
    useSelector((state) => state.esic.address[type].district)
  );
  const [street, setStreet] = useState(
    useSelector((state) => state.esic.address[type].street)
  );

  const [pincode, setPincode] = useState(
    useSelector((state) => state.esic.address[type].pincode)
  );

  useEffect(() => {
    dispatch(addESICAddress({ type: type, subtype: "state", val: geoState }));
  }, [geoState]);

  useEffect(() => {
    dispatch(addESICAddress({ type: type, subtype: "street", val: street }));
  }, [street]);

  useEffect(() => {
    dispatch(addESICAddress({ type: type, subtype: "pincode", val: pincode }));
  }, [pincode]);

  useEffect(() => {
    dispatch(
      addESICAddress({
        type: type,
        subtype: "district",
        val: district,
      })
    );
  }, [district]);

  useEffect(() => {
    if (geoState) {
      setDistricts(customData[geoState]);
      console.log(geoState);
    }
  }, [geoState]);

  switch (type) {
    case "present":
      var title = "Present";
      break;
    case "permanent":
      var title = "Permanent";
      break;
    case "nominee":
      var title = "Nominee";
      break;
  }

  return (
    <>
      <FormInput
        placeholder={title + " Street"}
        containerStyle={{ marginVertical: 10 }}
        // value={street}
        onChange={setStreet}
      />
      <DropDownForm
        placeholder={title + " State"}
        containerStyle={{ marginVertical: 10 }}
        value={geoState}
        setValue={setGeoState}
        data={states}
      />

      <DropDownForm
        placeholder={title + " District"}
        containerStyle={{ marginVertical: 10 }}
        value={district}
        setValue={setDistrict}
        data={districts}
      />

      <FormInput
        placeholder={title + " Pincode"}
        containerStyle={{ marginVertical: 10 }}
        // value={pincode}
        onChange={setPincode}
      />
    </>
  );
};

export default memo(AddressDropdown);
