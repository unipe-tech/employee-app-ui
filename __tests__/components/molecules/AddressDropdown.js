import { render, fireEvent, waitFor } from "@testing-library/react-native";
import { describe, expect, test } from "@jest/globals";
import AddressDropdown from "../../../components/molecules/AddressDropdown";

describe("AddressDropdown", () => {
  test("renders UI correctly", () => {
    const tree = render(<AddressDropdown />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
