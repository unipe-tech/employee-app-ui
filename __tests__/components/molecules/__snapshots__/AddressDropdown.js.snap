// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressDropdown renders UI correctly 1`] = `
Array [
  <View
    style={
      Object {
        "marginVertical": 10,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderColor": "#aaaaaa",
          "borderRadius": 5,
          "borderWidth": 1.5,
          "flexDirection": "row",
          "height": 50,
          "paddingHorizontal": 19.736842105263158,
          "width": "100%",
        }
      }
    >
      <TextInput
        autoCompleteType="off"
        editable={true}
        keyboardType="default"
        onChangeText={[Function]}
        placeholder="Present Street"
        placeholderTextColor="#808080"
        style={
          Object {
            "color": "#000000",
            "flex": 1,
            "fontFamily": "Montserrat-Regular",
            "fontSize": 14,
            "lineHeight": 18,
          }
        }
        value=""
      />
    </View>
  </View>,
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "marginVertical": 10,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderColor": "#aaaaaa",
            "borderRadius": 5,
            "borderWidth": 1.5,
            "flexDirection": "row",
            "height": 50,
            "paddingHorizontal": 19.736842105263158,
            "width": "100%",
          }
        }
      >
        <TextInput
          autoCompleteType="off"
          editable={false}
          keyboardType="default"
          onChangeText={[Function]}
          placeholder="Present State"
          placeholderTextColor="#808080"
          style={
            Object {
              "color": "#000000",
              "flex": 1,
              "fontFamily": "Montserrat-Regular",
              "fontSize": 14,
              "lineHeight": 18,
            }
          }
          value=""
        />
        <Text />
      </View>
    </View>
  </View>,
  <Modal
    animationType="none"
    deviceHeight={null}
    deviceWidth={null}
    hardwareAccelerated={false}
    hideModalContentWhileAnimating={false}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onRequestClose={[Function]}
    panResponderThreshold={4}
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    statusBarTranslucent={false}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
    transparent={true}
    visible={false}
  />,
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "marginVertical": 10,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderColor": "#aaaaaa",
            "borderRadius": 5,
            "borderWidth": 1.5,
            "flexDirection": "row",
            "height": 50,
            "paddingHorizontal": 19.736842105263158,
            "width": "100%",
          }
        }
      >
        <TextInput
          autoCompleteType="off"
          editable={false}
          keyboardType="default"
          onChangeText={[Function]}
          placeholder="Present District"
          placeholderTextColor="#808080"
          style={
            Object {
              "color": "#000000",
              "flex": 1,
              "fontFamily": "Montserrat-Regular",
              "fontSize": 14,
              "lineHeight": 18,
            }
          }
          value=""
        />
        <Text />
      </View>
    </View>
  </View>,
  <Modal
    animationType="none"
    deviceHeight={null}
    deviceWidth={null}
    hardwareAccelerated={false}
    hideModalContentWhileAnimating={false}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onRequestClose={[Function]}
    panResponderThreshold={4}
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    statusBarTranslucent={false}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
    transparent={true}
    visible={false}
  />,
  <View
    style={
      Object {
        "marginVertical": 10,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderColor": "#aaaaaa",
          "borderRadius": 5,
          "borderWidth": 1.5,
          "flexDirection": "row",
          "height": 50,
          "paddingHorizontal": 19.736842105263158,
          "width": "100%",
        }
      }
    >
      <TextInput
        autoCompleteType="off"
        editable={true}
        keyboardType="default"
        onChangeText={[Function]}
        placeholder="Present Pincode"
        placeholderTextColor="#808080"
        style={
          Object {
            "color": "#000000",
            "flex": 1,
            "fontFamily": "Montserrat-Regular",
            "fontSize": 14,
            "lineHeight": 18,
          }
        }
        value=""
      />
    </View>
  </View>,
]
`;
